server/src/main/java/org/elasticsearch/common/settings/FallbackSecureSettings.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.settings;
+
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.hash.MessageDigests;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * An implementation of secure settings that fall back to the YML settings.
+ *
+ * WARNING: this is a temporary class to be used only in the context of Stateless. It applies only to a predetermined set of secure
+ * settings mainly for repository settings. Other secure settings do not fall back to YML settings.
+ */
+public class FallbackSecureSettings implements SecureSettings {
+    private final SecureSettings secureSettings;
+    private final Settings settings;
+    private final Set<String> names;
+
+    private FallbackSecureSettings(Settings settings) {
+        if (DiscoveryNode.isStateless(settings) == false) {
+            throw new IllegalArgumentException("FallbackSecureSettings are supported only in stateless");
+        }
+
+        SecureSettings secureSettings = Settings.builder().put(settings, true).getSecureSettings();
+        this.secureSettings = secureSettings;
+        this.settings = Settings.builder().put(settings, false).build();
+
+        Stream<String> stream = settings.keySet()
```

Comment:
I think I would prefer to go down a route of having a `insecure.` prefix for all these settings. We can then remove that prefix when looking it up. But leave them in `settings`.

So we define a new 
```
AffixSetting<String> insecure = Setting.prefixKeySetting("insecure.", key -> Setting.simpleString(key,  Property.NodeScope))
```

that holds the insecure secure setting values. This way we avoid the filter for specific keys and we can make it a more generic mechanism. Still only enabled for stateless.

Here we'd iterate through all those insecure settings, remove the prefix and keep the names after removing that. When looking up a secure setting, we can ask `insecure.getConcreteSetting(key).get(settings)`.

Reply:
Hi @henningandersen , thanks for the idea! I can try it out and come back in this PR. Please note that this method will require to have newly-named settings for the repository settings (with the `insecure.` prefix) in the .yml file. They can be intermixed with regular repository settings (which do not have the `insecure.` prefix) if needed. So it can look like this for example:

```
insecure.s3.client.default.access_key: "..."
insecure.s3.client.default.secret_key: "..."
insecure.s3.client.default.session_token: "..."
s3.client.default.endpoint: "..."
```

When the switch is done to the new secure settings, the infrastructure will need to additionally rename the settings (to remove the `insecure.` prefix). So far I was trying an approach where this would not be needed, but if this is acceptable (it's minor effort I assume, so it should be acceptable), I also like it since we don't need to filter for specific secure settings here.

cc @tlrx to re-adjust expected naming of the settings

Reply:
@henningandersen I implemented your suggestion, it works pretty well! I am resolving this for now, but feel free to re-open if needed.

Reply:
> cc @tlrx to re-adjust expected naming of the settings

I like Henning's idea and the result looks good.